src/shogun/machine/IterativeLinearMachine.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+* This software is distributed under BSD 3-clause license (see LICENSE file).
+*
+* Authors: Shubham Shukla
+*/
+
+
+#include <shogun/lib/config.h>
+
+#include <shogun/lib/common.h>
+#include <shogun/machine/IterativeLinearMachine.h>
+
+using namespace shogun;
+
+CIterativeLinearMachine::CIterativeLinearMachine() : CLinearMachine()
+{
+ m_current_iteration = 0;
+ m_complete = false;
+ register_params();
+}
+
+CIterativeLinearMachine::~CIterativeLinearMachine()
+{
+}
+
+bool CIterativeLinearMachine::train(CDotFeatures* f, CLabels* l)
+{
+   init_model(f, l);
```

Comment:
you need to set the iteration counter to 0 here, and the completed flag to false